go/api/response_handlers.go

```diff
@@ -56,3 +54,14 @@ func handleBooleanResponse(response *C.struct_CommandResponse) bool {
 	defer C.free_command_response(response)
 	return bool(response.bool_value)
 }
+
+func handleMapResponse(response *C.struct_CommandResponse) map[string]string {
```

Comment:
This is still a Map<String, String>. At a later point we might need to change it to be be more generic.

Reply:
Note, the most generic map for glide is `Map<GlideString, ?>`, not `Map<?, ?>`. The map key is always a `GlideString` (`string | byte[]`)

Reply:
We would have to add specific handlers in Go to return different types of map objects. This method is only handling map[string]string type in Go whereas Rust would have generic handler for the map type.